### ISSUE TITLE
[BrowserKit] Cookie expiration date in the past causes exception

### ISSUE THREAD

**Issue by bt** [symfony/symfony#21459]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.2.2

I'm getting an exception whilst using Browser-Kit whereby, if the response returns a `Set-Cookie` expiration time that's in the past, it throws the an exception like the following:

```
Fatal error: Uncaught UnexpectedValueException: The cookie expiration time "-39600" is not valid.
```

An example of an HTTP header in my response:

```
Set-Cookie	ADRUM_BTs=R:54|s:p; expires=Wed, 31-Dec-1969 13:00:00 GMT; path=/; HttpOnly
```

According to [RFC 6265](https://tools.ietf.org/html/rfc6265#section-3.1):

`Finally, to remove a cookie, the server returns a Set-Cookie header with an expiration date in the past.`

So therefore, having a date in the past is expected behaviour and should result in the cookie being deleted from the cookie jar, I'm assuming.

**Comment by xabbuh** [comment]
This is related to (caused by the same code as) #15656.

**Comment by bt** [comment]
Oh sorry, didn't see that one.

Is this something that's being worked on? I can see it's been open since 2015.

If accepted as an issue, I can make a PR for it.

**Comment by xabbuh** [comment]
@bt I was looking into it. But if you want to, you are invited to take over. :)

**Comment by javiereguiluz** [comment]
I'm not closing this as a duplicated of #15656 ... but they are so closely related, that the Pull Request could fix both issues at the same time. Thanks!

**Comment by xabbuh** [comment]
@bt I could not reproduce your specific issue. The `-39600` timestamp works well for me. Which PHP version did you use?

**Comment by bt** [comment]
@xabbuh OSX, PHP 7 as installed by Homebrew:

```
PHP 7.0.7 (cli) (built: May 27 2016 11:13:44) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
```

**Comment by xabbuh** [comment]
@bt This seems to be an issue in PHP that was fixed with 7.0.9 (see https://bugs.php.net/bug.php?id=66836 and https://3v4l.org/QTH1C). By the way, you can work around this by using `0` as the timestamp which still is in the past. So I am going to close here as the exact issue you experience cannot be fixed in the BrowserKit component.

**Comment by javiereguiluz** [comment]
But please, let's fix the other related issue.

**Comment by xabbuh** [comment]
@bt Do you want to fix #15656 (it's an easy one to start contributing 😄)? Otherwise, I'll open a PR later today.

**Comment by bt** [comment]
@xabbuh Oops, got sidetracked doing other stuff at work 😆  I see you've already made the PR to fix it, sorry about that!
